terraform/modules/arm-builder-vm/virtual_machine.tf

```diff
@@ -0,0 +1,149 @@
+# SPDX-FileCopyrightText: 2024 Technology Innovation Institute (TII)
+#
+# SPDX-License-Identifier: Apache-2.0
+
+resource "azurerm_virtual_machine" "main" {
+  name                = var.virtual_machine_name
+  resource_group_name = var.resource_group_name
+  location            = var.location
+  vm_size             = var.virtual_machine_size
+
+  delete_os_disk_on_termination    = true
+  delete_data_disks_on_termination = false
+
+  network_interface_ids = [azurerm_network_interface.default.id]
+
+  storage_image_reference {
+    publisher = "Canonical"
+    offer     = "0001-com-ubuntu-server-jammy"
+    sku       = "22_04-lts-arm64"
+    version   = "latest"
+  }
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  os_profile {
+    computer_name = var.virtual_machine_name
+    # Unused, but required by the API. May not be root either
+    admin_username = "foo"
+    admin_password = "S00persecret"
```

Comment:
Great work Joonas! but why ps hardcoded

Reply:
copied from x86 builder so should be fine.